### ISSUE TITLE
Vulnerable packages found

### ISSUE THREAD

**Issue by fshmcallister** [WoTTsecurity/wott-io#204]
We found vulnerable packages on {devices}. These packages could be used by an attacker to either gain access to your node, or escalate permission. It is recommended that you address this at your earliest convenience. Run sudo apt-get update && sudo apt-get upgrade to bring your system up to date. Please note that there might be vulnerabilities detected that are yet to be fixed by the operating system vendor.

part of #198

**Comment by fshmcallister** [comment]
**TITLE**
Vulnerable packages found
> update or remove packages

**TL;DR**
Vulnerable packages found on [DEVICE/ENDPOINT(S)]. These packages may be exploitable by an attacker to gain access to your [DEVICE], or escalate permissions. Make sure packages are up to date by running `sudo apt-get update && sudo apt-get upgrade`. If package is known to still be compromised, we recommend removing or disabling it. Note, some vulnerabilities may have yet to be addressed by the operating system vendor.

**FAQ**
Packages are a key part of most Linux based operating systems software. Typically when downloading additional software onto Linux systems, it comes in pre-compiled packages that contain relevant information about updates, versions, and security. Inbuilt software therefore also has packages and these packages depend on the Linux distribution run. For example, Debian distributions have `DEB` packages and Red Hat has `RPM` packages with many other package systems being based off these two.

Over time, if not updated, these packages become vulnerable to security risks leaving them open to exploitation from external attackers. These operating system packages can then be used by an attacker to gain access to a [DEVICE/NODE], or its permissions and then escalate them. Fortunately, these packages are updated frequently. WoTT can detect an out-of-date or insecure package. Usually, it is only necessary to update the package or packages by running `sudo apt-get update && sudo apt-get upgrade` to obtain the most recent, secure version. However, it is also possible that a package has not had its security vulnerability resolved yet by the operating system vendor. In this case, it is safer to disable a particular package where possible.

**Comment by AlEsmail** [comment]
@fshmcallister - this reco action accompanies CVEs correct? If this is the case, let's make some mention of that as a general concept.

@vpetersson how shall we handle the fact that in the medium term we want to give instructions on a package by package basis.

**Comment by vpetersson** [comment]
> how shall we handle the fact that in the medium term we want to give instructions on a package by package basis.

Yeah, I forgot to remove one from the list. We will be removing these from RA for the time being as we are reworking this. Let's leave this one as is for now.

**Comment by fshmcallister** [comment]
does this apply to insecure services too?

**Comment by fshmcallister** [comment]
**TITLE**
Vulnerable packages found
> update or remove packages

**TL;DR**
Vulnerable packages have been found. These packages may be exploitable by an attacker to gain access to, or escalate permissions, on your servers. Make sure packages are up to date by running `sudo apt-get update && sudo apt-get upgrade`. If package is known to still be compromised, we recommend removing or disabling it. Note, some vulnerabilities may have yet to be addressed by the operating system vendor.

**FAQ**
Packages are a key part of most Linux based operating systems software. Typically when downloading additional software onto Linux systems, it comes in pre-compiled packages that contain relevant information about updates, versions, and security. Inbuilt software therefore also has packages and these packages depend on the Linux distribution run. For example, Debian distributions have `DEB` packages and Red Hat has `RPM` packages with many other package systems being based off these two.

Over time, if not updated, these packages become vulnerable to security risks leaving them open to exploitation from external attackers. These operating system packages can then be used by an attacker to gain access to your server, or its permissions and then escalate them. Fortunately, these packages are updated frequently. WoTT can detect an out-of-date or insecure package. Usually, it is only necessary to update the package or packages by running `sudo apt-get update && sudo apt-get upgrade` to obtain the most recent, secure version. However, it is also possible that a package has not had its security vulnerability resolved yet by the operating system vendor. In this case, it is safer to disable a particular package where possible.

**Comment by a-martynovich** [comment]
@vpetersson are you sure this is up to date?

**Comment by vpetersson** [comment]
@a-martynovich missed this one as I initially thought we would ignore it, but I've now updated it for the time being to the new format.
